### PR TITLE
Testing framework upgrade

### PR DESCRIPTION
**Pull request by narayanaditya95**
Changes: 
* Add django 1.8 to tox tests.
* Instruct TOX to run tests over versions of djangoCMS (3.0, 3.1, develop)
* Switch to djangocms-helper for automatic virtual project creation. Adds support for django 1.8.
* Transfer all tests from example project to django package.
* Instruct Travis to iterate over django and djangoCMS versions.

<s>Dont merge yet.</s>
This PR aims to allow testing of djangocms-cascade over different versions of django and djangoCMS.
The existing bs3demo would require a lot of modifications to work against both ```django==1.7``` and ```django==1.8```. To allow creation of virtual projects, I switched to ```djangocms-helper```.
All tests have been copied from bs3demo to cmsplugin_cascade package.
<s>All tests except the tests under HTTP pass** on ```django==1.7```.</s>
All tests fail under ```django==1.8``` (so I have added them to allowed_failures on Travis).
This PR would involve a lot of cleanup of both the HELPER_SETTING in test_settings.py and the bs3demo projects.

@jrief Please review.
<s>I am currently debugging the reason why HTTP tests fail under known environments.<s>

### CODE REVIEW

File: cmsplugin_cascade/tests/test_button.py [hunk 1]
```diff
@@ -34,4 +34,4 @@ def test_internal_link(self):
             'link': {'pk': page.id, 'model': 'cms.Page', 'type': 'cmspage'}, 'target': ''}
         model_instance = add_plugin(self.placeholder, BootstrapButtonPlugin, 'en', glossary=glossary)
         html = model_instance.render_plugin({})
-        self.assertHTMLEqual(html, '<a href="/"  class="btn btn-success">HOME</a>')
+        self.assertHTMLEqual(html, '<a href="/en/"  class="btn btn-success">HOME</a>')
```
Comment (narayanaditya95) [inline cmsplugin_cascade/tests/test_button.py]: As seen from Travis log, the output of this tests depends on the version of djangoCMS involved as it passes on some and fails on others. The redirect URL can be either '/en/' or '/en-us'

File: tox.ini [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
-    django17: django>=1.7,<1.8
-    py{26,27}: wsgiref
+    django16: django<1.7
+    django17: django<1.8
+    django18: django<1.9
+    cms30: https://github.com/divio/django-cms/archive/support/3.0.x.zip
+    cms31: https://github.com/divio/django-cms/archive/support/3.1.x.zip
+    cms32: https://github.com/divio/django-cms/archive/develop.zip
+    py27: wsgiref
     -rexamples/bs3demo/requirements.txt
```
Comment (narayanaditya95) [inline tox.ini]: TO DO: Divinding requirements into basic package requirements and specific bs3demo project requirements.

**Comment by narayanaditya95** [comment]
@jrief 
I hope the PR is good enough to be merged.
Tests now run well over many versions of django (from 1.6 to 1.7) and djangoCMS (from 3.0.x to 3.2.0dev1). However, test for django 1.6 with cms 3.0.x behave differently from others and all tests for django 1.8 fail.  
I hope to make a separate PR for django 1.8 migration too.  
The tests and the example demo project have been made independent. There still exists a lot fo test-related configurations in the example demo (and vice-versa).  
Infact, djangocms-helper can itself serve a virtual project for each version of Django, making the demo project redundant.  
Suggestions would be appreciated.
Thank you.

**Comment by jrief** [comment]
Great Pull Request!
Strange that Py2.7+Django-1.6+CMS-3.0 fails. This worked in my environment.
Do you have any idea why the button is rendered as ``<a class="btn btn-success" href="/en-us/">`` instead of ``<a class="btn btn-success" href="/en/">``? Can I set the default language explicitly?

Why did you move the tests into the main source tree? This is something I try to avoid. The reason is, that otherwise the testing code is installed with every version fetched from PyPI. But nobody needs them. Therefore, if you absolutely have to separate it from examples, please add it outside of ``djangocms-cascade/cmsplugin_cascade``.
The reason I keep the test folder inside examples is, that while developing I want to add as few additional directories to my Python search path.

**Comment by narayanaditya95** [comment]
> Do you have any idea why the button is rendered as ```<a class="btn btn-success" href="/en-us/">``` instead of ```<a class="btn btn-success" href="/en/">```? Can I set the default language explicitly?
I am afraid I have no answer for the peculiar behaviour.  
> Why did you move the tests into the main source tree? This is something I try to avoid.
Thank you for pointing that out. I have addressed it in the latest commit.

**Comment by narayanaditya95** [comment]
@jrief Test cases now pass on all versions of Django<1.8.

I have started working on Django 1.8 compatibilty. Should I create a new PR for the same or continue adding them this PR?

**Comment by jrief** [comment]
I'm going to merge this, because I wanted to release 0.6 soon.
Thanks for all this work, it a heavily appreciated.
Therefore please start with a new PR for Django-1.8.

**Comment by narayanaditya95** [comment]
@jrief Thank you for the merge.

**Pull request merged**